src/main/java/io/iworkflow/controller/ShortlistCandidatesController.java

```diff
@@ -0,0 +1,170 @@
+            );
+        }
+
+        final String workflowId = WorkflowUtil.buildShortlistWorkflowId(employerId, candidateId);
+
+        final ImmutableShortlistInput input = ImmutableShortlistInput.builder()
+                .employerId(employerId)
+                .candidateId(candidateId)
+                .build();
+
+        try {
+            // Set the timeout to 8 minutes because there is a 5-minute window before sending the email
+            client.startWorkflow(ShortlistWorkflow.class, workflowId, 8 * 3600, input);
+        } catch (final ClientSideException e) {
+            if (e.getErrorSubStatus() == ErrorSubStatus.WORKFLOW_ALREADY_STARTED_SUB_STATUS) {
+                return ResponseEntity.ok(String.format("Already running workflowId: %s", workflowId));
+            }
+            throw new RuntimeException(String.format("Failed to start the workflow %s", workflowId), e);
+        }
+
+        return ResponseEntity.ok(String.format("Started workflowId: %s", workflowId));
+    }
+
+    @PostMapping("/revoke_shortlist")
+    public ResponseEntity<String> revokeShortlist(
+            @RequestBody Map<String, Object> requestBody
+    ) {
+        final String employerId = requestBody.get("employerId").toString();
+        final String candidateId = requestBody.get("candidateId").toString();
+
+        final String workflowId = WorkflowUtil.buildShortlistWorkflowId(employerId, candidateId);
+
+        try {
+            client.signalWorkflow(ShortlistWorkflow.class, workflowId, "", ShortlistWorkflow.SIGNAL_REVOKE_SHORTLIST, null);
+        } catch (final ClientSideException e) {
+            if (e.getErrorSubStatus() == ErrorSubStatus.WORKFLOW_NOT_EXISTS_SUB_STATUS) {
+                return ResponseEntity.ok(String.format("No running workflow to revoke for %s", employerId + "-" + candidateId));
+            }
+            throw new RuntimeException(String.format("revokeShortlist failed with workflow %s", workflowId), e);
+        }
+
+        return ResponseEntity.ok(String.format("Revoked shortlist for %s", employerId + "-" + candidateId));
+    }
+
+    @GetMapping("/email_sent_timestamp")
+    public ResponseEntity<Long> getEmailSentTimestamp(
+            @RequestParam(defaultValue = "test-employer") String employerId,
+            @RequestParam(defaultValue = "test-candidate") String candidateId
+    ) {
+        final String workflowId = WorkflowUtil.buildShortlistWorkflowId(employerId, candidateId);
+
+        final ShortlistWorkflow rpcStub = client.newRpcStub(
+                ShortlistWorkflow.class,
+                workflowId,
+                ""
+        );
+
+        Long timestamp;
+        try {
+            timestamp = client.invokeRPC(rpcStub::getEmailSentTimestamp);
+        } catch (final ClientSideException e) {
+            if (e.getErrorSubStatus() == ErrorSubStatus.WORKFLOW_NOT_EXISTS_SUB_STATUS) {
+                timestamp = 0L;
```

Comment:
Nit: I think this should return 404 instead. returning 0 value will be confusing.